MekHQ/src/mekhq/gui/view/PersonViewPanel.java

```diff
@@ -748,27 +759,99 @@ private JPanel fillFamily() {
             firsty++;
         }
 
-        if (campaign.getCampaignOptions().useParentage() && person.hasChildren()) {
-            lblChildren1.setName("lblChildren1"); // NOI18N //$NON-NLS-1$
-            lblChildren1.setText(resourceMap.getString("lblChildren1.text")); //$NON-NLS-1$
-            gridBagConstraints = new GridBagConstraints();
-            gridBagConstraints.gridx = 0;
-            gridBagConstraints.gridy = firsty;
-            gridBagConstraints.fill = GridBagConstraints.NONE;
-            gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
-            pnlFamily.add(lblChildren1, gridBagConstraints);
+        if (campaign.getCampaignOptions().useParentage()) {
+            if (person.hasChildren()) {
+                lblChildren1.setName("lblChildren1"); // NOI18N //$NON-NLS-1$
+                lblChildren1.setText(resourceMap.getString("lblChildren1.text")); //$NON-NLS-1$
+                gridBagConstraints = new GridBagConstraints();
+                gridBagConstraints.gridx = 0;
+                gridBagConstraints.gridy = firsty;
+                gridBagConstraints.fill = GridBagConstraints.NONE;
+                gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
+                pnlFamily.add(lblChildren1, gridBagConstraints);
 
-            lblChildren2.setName("lblChildren2"); // NOI18N //$NON-NLS-1$
-            lblChildren2.setText(person.getChildList());
-            gridBagConstraints = new GridBagConstraints();
-            gridBagConstraints.gridx = 1;
-            gridBagConstraints.gridy = firsty;
-            gridBagConstraints.weightx = 1.0;
-            gridBagConstraints.insets = new Insets(0, 10, 0, 0);
-            gridBagConstraints.fill = GridBagConstraints.NONE;
-            gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
-            pnlFamily.add(lblChildren2, gridBagConstraints);
-            firsty++;
+                gridBagConstraints = new GridBagConstraints();
+                gridBagConstraints.gridx = 1;
+                gridBagConstraints.weightx = 1.0;
+                gridBagConstraints.insets = new Insets(0, 10, 0, 0);
+                gridBagConstraints.fill = GridBagConstraints.NONE;
+                gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
+
+                for (Person child : person.getChildList()) {
+                    gridBagConstraints.gridy = firsty;
+                    lblChildren2 = new JLabel();
```

Comment:
This should be a loop local variable, otherwise the field becomes the last child.

Reply:
By re-declaring it like this, it works correctly. The previous JLabel is already part of the pnlFamily, so it works well. I've tested it on a family with 2 children, and they link correctly. Looking at it did find a wasteful declaration, so I've cleaned that up.